### ISSUE TITLE
Socket vs Socket6

### ISSUE THREAD

**Issue by jobisoft** [techsneeze/dmarcts-report-parser#2]
Warning introduced by the ipv6 commit (https://github.com/techsneeze/imap-dmarcts/commit/c0bce692992444137fe5afcc64bfd6f7c51a9911)

Constant subroutine main::AF_INET6 redefined at ...
Prototype mismatch main::AF_INET6 () vs none at ...

In the current code, you inlcude everything from Socket and just (inet_ntop inet_pton AF_INET6) from Socket6. 

If you remove that restriction and do

use Socket;
use Socket6;

The warning goes away. I HAVE ABSOLUTLY NO IDEA about WHY this works, and what really is up with socket and socket6. I just want to bring it to your attention.

**Comment by techsneeze** [comment]
I'm not able to duplicate that warning on my system.  What versions of perl and modules do you have?

**Comment by jobisoft** [comment]
Perl is v5.20.2
libio-socket-ip-perl is v0.32-1
libio-socket-inet6-perl is v2.72-1

up-to-date Debian 8 install

**Comment by techsneeze** [comment]
Oddly it runs fine on:
perl, v5.8.8
perl-IO-Socket-INET6-2.51-2
perl-Socket6-0.19-3

The original contributor presumably didn't have any warnings either.

Regardless, I've been able to replicate the warnings on a different machine (Debian Jessie).

References to the same warning bring up bugs from 2008, which have since been closed.  It makes me wonder if it is a regression or something.  We can make the suggested change, but it would be nice to understand "why" it was.
